src/components/OpenInMapsActionSheet.js

```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { View, Text, StyleSheet, Modal, ListView } from 'react-native';
+import ActionSheet from 'react-native-actionsheet';
+
+export default class OpenInMapsActionSheet extends Component {
+  static propTypes = {
+    apps: PropTypes.array,
+    open: PropTypes.bool,
+    title: PropTypes.string,
+    onPress: PropTypes.func.isRequired,
+  }
+
+  static defaultProps = {
+    apps: [],
+    open: false,
+    title: 'Open Location',
+  }
+
+  constructor(props, context) {
+    super(props, context);
+
+    this.state = {
+      cancelIndex: props.apps.length,
+    };
+  }
+
+  handlePress = (index) => {
+    this.props.onPress(index);
+  }
+
+  componentDidMount() {
+    if (this.props.open && this.ActionSheet) {
+      this.ActionSheet.show();
+    }
+  }
+
+  componentWillUnmount() {
+    this.ActionSheet = null;
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({ cancelIndex: nextProps.apps.length });
```

Comment:
I don't think `cancelIndex` needs to be a state. Couldn't it just be just an attribute of this object? (i.e. `this._cancelIndex`)?